share/spack/gitlab/cloud_pipelines/stacks/ml/spack.yaml

```diff
@@ -0,0 +1,136 @@
+spack:
+  view: false
+
+  concretizer:
+    reuse: false
+    unify: false
+
+  config:
+    concretizer: clingo
+    install_tree:
+      root: /home/software/spack
+      padded_length: 384
+      projections:
+        all: "{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}"
+
+  packages:
+    all:
+      compiler: [gcc@11.2.0]
+      target: [x86_64_v4]
+      variants: +cuda cuda_arch=80
+
+  specs:
+    # Horovod
+    - py-horovod
+
+    # JAX
+    - py-jax
+    - py-jaxlib
+
+    # Keras
+    - py-keras
+    - py-keras-applications
+    - py-keras-preprocessing
+    - py-keras2onnx
+
+    # PyTorch
+    - py-botorch
+    - py-efficientnet-pytorch
+    - py-gpytorch
+    - py-kornia
+    - py-pytorch-gradual-warmup-lr
+    - py-pytorch-lightning
+    - py-segmentation-models-pytorch
+    - py-timm
+    - py-torch
+    - py-torch-cluster
+    - py-torch-scatter
+    - py-torch-sparse
+    - py-torch-spline-conv
+    - py-torchaudio
+    - py-torchdata
+    - py-torchfile
+    - py-torchgeo
+    - py-torchmeta
+    - py-torchmetrics
+    - py-torchtext
+    - py-torchvision
+    - py-vector-quantize-pytorch
+
+    # scikit-learn
+    - py-scikit-learn
+    - py-scikit-learn-extra
+
+    # TensorBoard
+    - py-tensorboard
+    - py-tensorboard-data-server
+    - py-tensorboard-plugin-wit
+    - py-tensorboardx
+
+    # TensorFlow
+    - py-tensorflow
+    - py-tensorflow-hub
+    - py-tensorflow-probability
+
+    # XGBoost
+    - py-xgboost
+    - xgboost
+
+  mirrors: { "mirror": "s3://spack-binaries/develop/ml" }
+
+  gitlab-ci:
+    script:
+      - . "./share/spack/setup-env.sh"
+      - spack --version
+      - cd ${SPACK_CONCRETE_ENV_DIR}
+      - spack env activate --without-view .
+      - spack config add "config:install_tree:projections:${SPACK_JOB_SPEC_PKG_NAME}:'morepadding/{architecture}/{compiler.name}-{compiler.version}/{name}-{version}-{hash}'"
+      - mkdir -p ${SPACK_ARTIFACTS_ROOT}/user_data
+      - if [[ -r /mnt/key/intermediate_ci_signing_key.gpg ]]; then spack gpg trust /mnt/key/intermediate_ci_signing_key.gpg; fi
+      - if [[ -r /mnt/key/spack_public_key.gpg ]]; then spack gpg trust /mnt/key/spack_public_key.gpg; fi
+      - spack -d ci rebuild > >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_out.txt) 2> >(tee ${SPACK_ARTIFACTS_ROOT}/user_data/pipeline_err.txt >&2)
+
+    image: "ghcr.io/spack/e4s-amazonlinux-2:v2022-03-21"
+
+    mappings:
+      - match:
+          - llvm
+        runner-attributes:
+          tags: [ "spack", "huge", "x86_64" ]
```

Comment:
But these tags only request `x86_64`.  At first I assumed this was just *recommended* for your PR, and wasn't likely to be the related to your build failures, because all our aws runners, and I thought all the UO runners too, were able to build for all levels of optimization.  But I found some UO runners are only tagged up to `x86_64_v3`, and sometimes those runners picked up build jobs for this PR.  Over [here](https://github.com/spack/spack/pull/32498), I attempted to validate this before bringing it up, and it's looking like it could actually have been the cause of these intermittent build failures.

Change `x86_64` to `x86_64_v4` here, and below, I think it should do the trick.